include/libdnf/rpm/repo_sack.hpp

```diff
@@ -34,14 +34,36 @@ class Base;
 
 namespace libdnf::rpm {
 
-class RepoSack : public libdnf::sack::Sack<Repo, RepoQuery> {
+class RepoSack : public sack::Sack<Repo, RepoQuery> {
 public:
     explicit RepoSack(Base & base) : base(&base) {}
 
     /// Creates new repository and add it into RepoSack
     RepoWeakPtr new_repo(const std::string & id, Repo::Type type = Repo::Type::AVAILABLE);
 
+    /// Creates new repositories according to the configuration in the file defined by path.
+    /// The created repositories are added into RepoSack.
+    void new_repos_from_file(const std::string & path);
+
+    /// Creates new repositories according to the configuration in the file defined by "config_file_path"
```

Comment:
Maybe the functionality could be with function that merge the main config. It will reduce the number of parsings. What about to add TODO?

Reply:
"config_file_path" contains the main configuration, but may also contain the rpm repository definition. It is parsed 2 times. Do we want an optimized interface for parsing everything in "config_file_path" in one pass? Or prefer a separate code?
I will add TODO.